### ISSUE TITLE
Reading System Preferences

### ISSUE THREAD

**Issue by alex-cory** [sindresorhus/electron-util#25]
Currently there's `openSystemPreferences`, what if there was a `readSystemPreferences` or something similar. That way we could see what system preferences were currently set.

**Comment by sindresorhus** [comment]
Yeah, could be useful, but there's no "one way" to read the system preferences. What preferences specifically do you need?

**Comment by alex-cory** [comment]
So, I specifically need to display
`Accessibility` has been enabled for the app ([this might work for Accessibility](https://github.com/4ndv/node-mac-accessibility-features-check))
`Full Disk Access` has been enabled for the app
`Files and Folders` has `Documents`, `Downloads`, and `Desktop` enabled for the app

**Comment by sindresorhus** [comment]
It used to be possible to read the database of permissions directly, but this is no longer possible: https://apple.stackexchange.com/questions/362865/macos-list-apps-authorized-for-full-disk-access

Here's how we could do this:

1. https://github.com/karaggeorge/macos-accessibility-permissions
2. We can try reading a certain off-limits file: https://github.com/MacPaw/PermissionsKit/blob/43744ea29edb296e177d5890f9fe50d3f5476d52/PermissionsKit/Private/FullDiskAccess/MPFullDiskAccessAuthorizer.m#L68-L76
3. We can check whether those directories are writable. `fs.accessSync(path, fs.constants.W_OK);`.

**Comment by alex-cory** [comment]
How do you think this api should look?
```js
readSystemPreferences('Accessibility')
readSystemPreferences('FullDiskAccess')
readSystemPreferences('FilesAndFolders', 'Documents')
readSystemPreferences('FilesAndFolders', 'Downloads')
readSystemPreferences('FilesAndFolders', 'Desktop')
```

**Comment by sindresorhus** [comment]
@alex-cory They should be separate methods as they all can't behave exactly the same (some might need to be async, for example).

```js
const {hasPrivacyPermission} = require('electron-util');

console.log(hasPrivacyPermission.fullDiskAccess());
console.log(hasPrivacyPermission.filesAndFoldersDesktop());
```

**Comment by alex-cory** [comment]
I can do my best to help implement this.

**Comment by SinghManmohan** [comment]
@sindresorhus : Can we open Notifications settings from System Preference directly?
```
const {openSystemPreferences} = require('electron-util');

openSystemPreferences('notifications
<img width="662" alt="Screen Shot 2021-02-03 at 2 55 47 PM" src="https://user-images.githubusercontent.com/29338074/106820260-e831b580-662f-11eb-8674-c2c46cb89fbd.png">
');
```

**Comment by SinghManmohan** [comment]
@sindresorhus  : Please ignore.. i just tried a sample app and it works.. my bad!
